com.unity.ml-agents/Runtime/Actuators/IActuator.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.MLAgents.Inference;
+using Unity.MLAgents.Policies;
+using UnityEngine;
+
+namespace Unity.MLAgents.Actuators
+{
+    public readonly struct ActuatorSpace
+    {
+        public SpaceType ActuatorSpaceType =>
+            ContinuousActionSpaceDef.NumActions > 0 ? SpaceType.Continuous : SpaceType.Discrete;
+        public ActionSpaceDef ContinuousActionSpaceDef { get; }
+        public ActionSpaceDef DiscreteActionSpaceDef { get; }
+
+        public ActuatorSpace(ActionSpaceDef continuousActionSpaceDef,
+            ActionSpaceDef discreteActionSpaceDef)
+        {
+            ContinuousActionSpaceDef = continuousActionSpaceDef;
+            DiscreteActionSpaceDef = discreteActionSpaceDef;
+        }
+
+    }
+    /// <summary>
+    /// Abstraction that facilitates the execution of actions.
+    /// </summary>
+    public interface IActuator : IActionReceiver
+    {
+        int TotalNumberOfActions { get; }
+
+        ActuatorSpace ActuatorSpace { get; }
```

Comment:
Why do we need ActuatorSpace? ActionSpaceDef can handle both types in the future, right?

Reply:
yeah, looks like i can kill it

Reply:
I was using as the holders for the different actuator spaces

Reply:
yes, but there are no discrete branches in the continuous case which is what this is for.